types/chart.js/index.d.ts

```diff
@@ -801,6 +803,20 @@ declare namespace Chart {
         lazy?: boolean;
         easing?: Easing;
     }
+
+    interface DoughnutModel {
```

Comment:
Was this intended to be a subtype of Model?  If not, it seems like the properties that are on `Model` but not `DoughnutModel` will be unavailable on `ModelType`.

Reply:
Good catch! These would be major breaking changes as they are currently written, which I didn't realize at first.

I wasn't intending for `DoughnutModel` to be a subtype of `Model`. Each chart model seemed very different looking at the chart.js code. I was hoping these models would be very explicit on what properties are going to be defined to avoid a bunch of optional properties or base properties defined that would never have values. For example, I didn't want to just add `circumference?: number;` to the `Model` definition because when I get the data for the Doughnut chart I know it is always defined.

Any suggestions on how to proceed?

I was thinking that the model and view definitions would eventually turn into something like `_model: BarModel | BubbleModel | DoughnutModel | LineModel | PolarAreaModel | RadarModel;` long-term. That was my mindset when creating this change. I don't have the time to do that right now so I just focused on getting the DoughnutModel updated on this PR.

Reply:
How would a consumer know which of the XModel types to use?  Without that information, they'd only have access to the properties common to all of them.

Reply:
Consumers would need to use `as` to use assertion to one of the specific types of models. Otherwise, yes they would just get the common properties across all union types. That might not be a good way to do this because consumers might not know they need to do that even if the names of the models match the chart types.

Is there a better way to add these missing properties from the Doughnut chart without making everything optional? When I create a Doughnut chart, I know the chart's data will be in a specific format so I was hoping I could reflect that in these definitions somehow.

Reply:
One common way to handle this would be to have a `kind` property on the model that tells you which set of properties are available.  I take it the underlying library doesn't have anything like that.  If you were consuming `_model` in pure JS, how would you know which properties to look at?

Reply:
The chart.js library seems to define a completely different `_model` structure for each type of chart it can render. It seems like inherit knowledge of the type of chart being rendered is needed for the `_model` property to be of any use to consumers.

* [Bubble _model](https://github.com/chartjs/Chart.js/blob/master/src/controllers/controller.bubble.js#L140-L151)
* [Doughnut _model](https://github.com/chartjs/Chart.js/blob/master/src/controllers/controller.doughnut.js#L242-L257)
* [Line _model](https://github.com/chartjs/Chart.js/blob/master/src/controllers/controller.line.js#L122-L135)

No common `kind` (or the equivalent) property seems to exist in the JS form. If that did exist, doing the discriminated unions would be possible.

Reply:
Personally, I haven't seen a good solution to this problem.  Options including loading `Model` with an optional declaration of every possible property, or cobbling together something like 

```ts
interface Model1 {
    P1: string;
    P2: number;
    P3?: never;
}
interface Model2 {
    P1?:never;
    P2: number;
    P3: boolean;
}

declare const _model: Model1 | Model2;
```

`_model` will have `P1` and `P3` optional and `P2` required and the declaration accurately conveys that the caller should probably assert one of the individual model types.  The downside is that all the declarations have to stay in sync (there might be something clever you can do with index signatures, but I didn't find it while experimenting).

Reply:
I think I'll pursue the optional declaration for every possible property on `Model` to not break existing users and then possibly still export `DoughnutModel` for users that want to assert to that model type. I don't like the other option of adding a bunch of `never` properties for the `DoughnutModel`. That seems to unnecessarily complicate that model and keeping everything in sync seems difficult.

Reply:
Understandable.  Sorry I didn't have a better answer for you.

Reply:
That's okay. Hopefully as these definitions progress, we can come up with a better solution for this scenario.

Thanks for all the help with this PR. I really appreciate the discussion.